File: kotlin/src/main/kotlin/sum_of_multiples/SumOfMultiples.kt
package sum_of_multiples

internal object SumOfMultiples {

    fun sum(factors: Set<Int>, limit: Int): Int {
        if (factors.isEmpty()) return 0
        val include = Array(limit) { false }
        for (f in factors) if (f in 1..limit) for (i in f until limit step f) {
            include[i] = true
        }
        return include.withIndex().sumOf { if (it.value) it.index else 0 }
    }

}

File: python/sum-of-multiples/sum_of_multiples.py
def sum_of_multiples(limit: int, multiples: list[int]) -> int:
    _multiples = set()
    for num in multiples:
        if num == 0:
            continue
        for multiple in range(num, limit, num):
            _multiples.add(multiple)
    return sum(_multiples)
